xnmt/translator.py

```diff
@@ -277,9 +307,16 @@ def __call__(self, translator, dec_state, src, trg):
       except ValueError:
         pass
       # Calculate the evaluation score
-      eval_score.append(self.evaluation_metric.evaluate_fast(trg_i.words, sample_i))
+      score = self.evaluation_metric.evaluate_fast(trg_i.words, sample_i)
+      eval_score.append(0 if math.isnan(score) else score)
+
+    true = dy.inputTensor(eval_score, batched=True)
+    loss_exp = self.regressor.calc_loss(decode_hidden_state, eval_score, dim)
+    loss_exp.backward()
+    self.regressor.trainer.update()
```

Comment:
There shouldn't be any training update or backward in calc_loss function

Reply:
Oh, I thought this would train the Regressor according to the inputs. How do you suggest that t=should be done?

Reply:
Okay, got it. I shall look into the segmenting Encoder as suggested by you. Thanks.